src/Utils/CallbacksArrayReducer.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace loophp\collection\Utils;
+
+use Closure;
+use Iterator;
+
+/**
+ * @immutable
+ *
+ * @template TKey
+ * @template T
+ *
+ * phpcs:disable Generic.Files.LineLength.TooLong
+ */
+final class CallbacksArrayReducer
```

Comment:
We could shorten the name to `CallbacksReducer`. Due to the plural it's quite clear we are dealing with a list of multiple callbacks

Reply:
Oops, too late, we can tackle that in a follow-up PR...